dbt_platform_helper/providers/config.py

```diff
@@ -0,0 +1,105 @@
+import os
+from copy import deepcopy
+from pathlib import Path
+
+import click
+import yaml
+from schema import SchemaError
+from yaml.parser import ParserError
+from yamllint import linter
+from yamllint.config import YamlLintConfig
+
+from dbt_platform_helper.constants import PLATFORM_CONFIG_FILE
+from dbt_platform_helper.providers.platform_config_schema import PlatformConfigSchema
+from dbt_platform_helper.utils.messages import abort_with_error
+
+
+class ConfigProvider:
+    def __init__(self, config_validator, config=None, echo=click.secho):
+        self.config = config or {}
```

Comment:
still think having this instance attribute confuses matters especially as setting it in the constructor at best does nothing since we don't have a getter for it and don't reference it directly anywhere in our code, and at worst causes confusion and bugs because someone thinks they've set it and then it gets overridden when we call the load function to access it.